test/dict.jl

```diff
@@ -787,6 +787,20 @@ end
           [v for (k, v) in d] == [d[x[1]] for (i, x) in enumerate(d)]
 end
 
+@testset "consistency of dict iteration order (issue #56841)" begin
+    d = Dict(randn() => randn() for _ = 1:100)
+    @test [k for (k,v) = d] ==
+        [k for k = keys(d)] ==
+        [k for (k,v) = pairs(d)] ==
+        collect(keys(d))
```

Comment:
Do these need to be `collect`ed or is a check with `any`/`all` (consuming the iterators in the process) also good here? Would save on some allocations.

Reply:
I don't think we should worry about optimizing allocations here; this code is run once with n=100 so it will be compile-time dominated anyway.

Reply:
> How exactly?

Something like this:

```julia
@test all(zip(d, keys(d), pairs(d))) do (kv, k, kpair)
   first(kv) == k == first(kpair)
end
```

(might need some adjustment in the argument destructuring of the `do` block - not 100% sure this unpacks correctly)